### PR TITLE
GRF-775 create add template attribute modal

### PR DESCRIPTION
**Pull request by amigaire**
**Description (for Contributor and Core Developer)**

This PR only contains:
- add button to open modal
- modal opens and display. Cancel button works, create button simulate a success call to add attribute
- template customization feature flag is taken into account

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

**Review by antoinetrouve — APPROVED** [review]

File: src/Akeneo/Category/front/src/configuration/Page.tsx [hunk 1]
```diff
@@ -127,7 +127,7 @@ const Page: FC = () => {
                       )
                   }
               >
-                Enriched Categories
+                Enriched Categories: Customize Template
```
Comment (BryanS29-akeneo) [inline src/Akeneo/Category/front/src/configuration/Page.tsx:130]: not translated text? is it temporary?

Comment (amigaire) [reply src/Akeneo/Category/front/src/configuration/Page.tsx:130]: this is for the configuration page of the micro frontend, I don't know if we should translate or not

**Review by BryanS29-akeneo — APPROVED** [review]

File: src/Akeneo/Category/front/src/feature/components/templates/AddTemplateAttributeModal.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+const HelperField = styled(Helper)`
+  width: 100%;
+`;
+
+type Form = { label: string; code: string };
+type FormError = { label: string[]; code: string[] };
+
+type Props = {
+    template_id: string;
```
Comment (pchasle) [inline src/Akeneo/Category/front/src/feature/components/templates/AddTemplateAttributeModal.tsx:27]:
```suggestion
    templateId: string;
```
camelCase for typescript code

File: src/Akeneo/Category/front/src/feature/components/templates/AddTemplateAttributeModal.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
+import {translate, useTranslate} from '@akeneo-pim-community/shared';
+import {useState} from "react";
+import {useCreateAttribute} from "../../hooks/useCreateAttribute";
+import {userContext} from "@akeneo-pim-community/shared/lib/dependencies/user-context";
+import {TextInput} from "akeneo-design-system/lib/components/Input/TextInput/TextInput";
+import {AttributesIllustration, Button, Field, Helper, Link, Modal} from "akeneo-design-system";
```
Comment (pchasle) [inline src/Akeneo/Category/front/src/feature/components/templates/AddTemplateAttributeModal.tsx:6]: Your IDE auto-import configuration is not doing a good work :thinking:
It shouldn't import files from deep into the library folders.
```suggestion
import {AttributesIllustration, Button, Field, Helper, Link, Modal, TextInput} from "akeneo-design-system";
```

File: src/Akeneo/Category/front/src/feature/pages/TemplatePage.tsx [hunk 4]
```diff
@@ -19,6 +19,9 @@ import {EditTemplatePropertiesForm} from '../components/templates/EditTemplatePr
 import {TemplateOtherActions} from '../components/templates/TemplateOtherActions';
 import {useCategoryTree, useTemplateByTemplateUuid} from '../hooks';
 import {Template} from '../models';
+import {AddTemplateAttributeModal} from "../components/templates/AddTemplateAttributeModal";
+import {Button} from "akeneo-design-system/lib/components/Button/Button";
```
Comment (pchasle) [inline src/Akeneo/Category/front/src/feature/pages/TemplatePage.tsx:23]:
```suggestion
```
Should be added to the 'akeneo-design-system' import

File: src/Akeneo/Category/front/src/feature/pages/TemplatePage.tsx [hunk 5]
```diff
@@ -148,6 +156,16 @@ const TemplatePage: FC = () => {
           >
             {translate('pim_common.properties')}
           </TabBar.Tab>
+          {featureFlags.isEnabled('category_template_customization') && (
```
Comment (pchasle) [inline src/Akeneo/Category/front/src/feature/pages/TemplatePage.tsx:159]: Looking at Figma, I think that the button should be in the `EditTemplateAttributesForm` component in the `SectionTitle`

Comment (amigaire) [reply src/Akeneo/Category/front/src/feature/pages/TemplatePage.tsx:159]: Will be done in the next PR https://github.com/akeneo/pim-community-dev/pull/19443

**Review by pchasle — APPROVED** [review]

**Pull request merged**